src/js/stores/DCOSStore.js

```diff
@@ -35,36 +37,53 @@ class DCOSStore extends EventEmitter {
     this.proxyListeners = [
       {
         event: MARATHON_GROUPS_CHANGE,
-        handler: this.processMarathonData,
+        handler: this.processMarathonGroups,
         store: MarathonStore
       },
       {
         event: MESOS_SUMMARY_CHANGE,
-        handler: this.processMesosData,
+        handler: this.processMesosStateSummary,
         store: MesosSummaryStore
       }
     ];
   }
 
   /**
-   * Process Marathon data
+   * Fetch service version/configuration from Marathon
+   * @param {string} serviceID
+   * @param {string} versionID
+   */
+  fetchServiceVersion(serviceID, versionID) {
+    MarathonStore.fetchServiceVersion(serviceID, versionID);
+  }
+
+  /**
+   * Fetch service versions/configurations from Marathon
+   * @param {string} serviceID
+   */
+  fetchServiceVersions(serviceID) {
+    MarathonStore.fetchServiceVersions(serviceID);
+  }
+
+  /**
+   * Process Marathon groups data
    * @param {ServiceTree} data
    */
-  processMarathonData(data) {
+  processMarathonGroups(data) {
```

Comment:
I'd argue `processMarathonServiceTree` would be more explicit, given that it definitely expects a `ServiceTree`?